0000-experiment-dataframe.md

```diff
@@ -0,0 +1,320 @@
+# combine trend data
+...
+```
+
+Analysis results (combined):
+
+|          | name   | value  | quality | device_components |  backend        | creation_datetime          | experiment | ... |
+|----------|--------|--------|---------|-------------------|-----------------|----------------------------|------------|-----|
+| 03fe91s1 | T1     | 100e-6 | good    | [Q0]              |  ibm_washington | 2022-11-22 04:01:30.313872 | T1         | ... |
+| 1c4ad2a1 | T1     | 103e-6 | good    | [Q0]              |  ibm_washington | 2022-11-22 05:01:14.123201 | T1         | ... |
+
+
+#### Complement
+
+As these examples show, experimentalist can distinguish the outcomes of batch (with `experiment`) and parallel experiments (with `device_components`). Even though there are multiple entries for the identical parameter (example 3), they have different result IDs and creation date times. In other words, one can sort the entries by `creation_datetime` and get the time-ordered sequence of `value` for the trend analysis:
+
+```python
+sorted_frame = result_dataframe.sort_values(by=["creation_datetime"])
+```
+
+alternatively one can directly visualize the data frame with seaborn:
+
+```python
+import seaborn as sns
+
+sns.relplot(
+    data=result_dataframe,
+    kind="line",
+    x="creation_datetime",
+    y="value",
+)
+```
+
+See pandas documentation for more details.
+
+### B. Introduction of artifacts
+
+Result database allows experimentalists to store `artifact` data per experiment ID. Thus, in QE, we can add `ExperimentData.artifact` to save supplementary data such as raw curve data points. In principle this can be a free-from dictionary, but it would be better to provide the artifact base class and subclasses to support custom JSON serialization to generate the REST API payload and the input data type validation. In the following we describe `CurveAnalysisArtifact` for `CurveAnalysis`. Note that artifact must be the [thread-safe container](https://github.com/Qiskit/qiskit-experiments/blob/b150427f86c74a02d638e06507864c7bb060c27c/qiskit_experiments/database_service/utils.py#L168-L245) subclass. Each analysis class must know the associated artifact subclass to populate the supplementary data.
+
+The artifact is a part of `ExpeirmentData`, however, each `BaseAnalysis._run_analysis` method must instantiate and return a list of artifact instances to the experiment data. For example, a composite analysis instance may return multiple artifacts for each qubit or experiment.
+
+
+#### BaseArtifact
+
+The base class of the artifact. Probably Python `dataclass`. This class implements `__json_encode__` and `__json_decode__` methods and subclass can override if necessary.
+This base class may have the field for list of figures. In addition to this, artifact object must provide metadata including
+- experiment (str): Name of experiment that generated this data.
+- device_components (List[DeviceComponent]): List of device components that this data is associated with.
```

Comment:
I don't think this is necessary for a generic artifact. Probably you just need something like `name` `artifact_id` `experiment_id`, `created`, maybe something like `filename/format/file_format` would be useful too for downloading directly from a db for things like images, or csv files, json etc

Reply:
This must be resolved by redesigning the data structure. We can create artifact entry per data rather than per analysis. https://github.com/Qiskit/rfcs/pull/28/commits/263e94a3c5a02e3be14c7916652590b63e970ba7